requests_cache/backends/base.py

```diff
@@ -212,20 +212,31 @@ class BaseStorage(MutableMapping, ABC):
         serializer: Custom serializer that provides ``loads`` and ``dumps`` methods
     """
 
+    DEFAULT_SERIALIZER = 'pickle'
+
     def __init__(
         self,
-        secret_key: Union[Iterable, str, bytes] = None,
-        salt: Union[str, bytes] = None,
-        suppress_warnings: bool = False,
         serializer=None,
+        suppress_warnings: bool = False,
         **kwargs,
     ):
-        self.serializer = init_serializer(serializer, secret_key=secret_key, salt=salt)
-        logger.debug(f'Initializing {type(self).__name__} with serializer: {self.serializer}')
+        serializer = serializer or self.DEFAULT_SERIALIZER
+
+        if "secret_key" in kwargs and serializer == 'pickle':
```

Comment:
While I completely agree with the separation-of-concerns point, there's something else going on here. 

First, using a keyword to select a serializer (`self.serializer = SERIALIZERS[serializer]`) probably can't be ditched.

Second, the parts with `callable` and `secret_key` are all about example compatibility. The logic here is really all about the example in [Security](./docs/security.rst). In particular, the fact that it allows a user to pass the keyword "secret_key," which changes a pickle serializer to safe_pickle, and passes that keyword into the safe_pickle serializer. The most straightforward fix would be to change that example to:

```python

>>> from requests_cache import CachedSession
>>> from requests_cache.serializers import safe_pickle_serializer
>>> session = CachedSession(serializer=safe_pickle_serializer(secret_key="hunter2"))
>>> session.get('https://httpbin.org/get')
```

This would, however, be a breaking change for anyone who relied on the prior example in their code. The logic code in `backend.py` is essentially a band-aid to make that example keep working, as well as anyone else who was relying on handed-down keyword arguments from `CachedSession` to the serializer.

The alternative would be to modify the SerializerPipeline to have a `__call__` function which would then pass any keyword arguments into any callable stages (e.g. any classes that can be instantiated as objects). This might not be a bad idea, and would allow the same hand-down behavior to extend to SerializerPipeline. But this would break using simple classes with dumps / loads (e.g. calling `pickle` would throw an exception). 

I'm open to either solution on that one, but it would make SerializerPipeline a bit messier.

But I think we can delete the `is_serializer` check, as an appropriate error will be thrown the first time the serializer is used. This is fixed in e9b6b414d6b356436032df634f7ee8e5ecc50d8c

Reply:
> This would, however, be a breaking change for anyone who relied on the prior example in their code. The logic code in backend.py is essentially a band-aid to make that example keep working, as well as anyone else who was relying on handed-down keyword arguments from CachedSession to the serializer.

Gotcha. Since that was added fairly recently, and there's a better general solution now, I think it would be reasonable to deprecate that for 0.7 and remove it in a subsequent release. I can handle that later.